### PR TITLE
KA: Engolo implemented

### PR DESCRIPTION
**Pull request by msbeck**
Copied laamb and changes values for engolo

### CODE REVIEW

**Comment by Saintis** [comment]
I'd be tempted to refactor Engolo and Laamb functionality into a helper function called `wrestling-breaker` or similar.

**Comment by Saintis** [comment]
Good stuff!

File: src/clj/game/cards/icebreakers.clj [hunk 1]
```diff
@@ -52,6 +52,29 @@
                               :approach-ice breaker-auto-pump }
                              (:events cdef))))
 
+(defn- wrestling-breaker
+  "Laamb and Engolo. Makes currently encountered ice gain chosen type until end of encounter."
+  [cost ice-type]
+  {:once :per-turn
+   :cost [:credit cost]
+   :label (str "Turn currently encountered ice into " ice-type)
+   :msg (msg "turn " (:title current-ice) " into " ice-type)
```
[the remark this answers is not in the document]
Comment (NoahTheDuke) [reply src/clj/game/cards/icebreakers.clj]: I'm fixing that up!

**Comment by NoahTheDuke** [comment]
Fixed `wrestling-breaker` label and msg:
```clojure
   :label (str "Make currently encountered ice gain " ice-type)
   :msg (msg "make " (:title current-ice) " gain " ice-type)
```
Test:
```clojure
(deftest wrestling-breakers
  ;; Engolo, Laamb - Check ability to give a piece of ice a specific subtype
  (do-game
    (new-game (default-corp ["Ichi 1.0"])
              (default-runner ["Laamb" "Engolo"]))
    (play-from-hand state :corp "Ichi 1.0" "HQ")
    (take-credits state :corp)
    (core/gain state :runner :credit 100)
    (play-from-hand state :runner "Laamb")
    (play-from-hand state :runner "Engolo")
    (let [ichi (get-ice state :hq 0)
          laamb (get-program state 0)
          engolo (get-program state 1)]
      (run-on state "HQ")
      (core/rez state :corp ichi)
      (is (not (core/has-subtype? (refresh ichi) "Barrier")) "Ichi shouldn't start with Barrier subtype")
      (is (not (core/has-subtype? (refresh ichi) "Code Gate")) "Ichi shouldn't start with Code Gate subtype")
      (card-ability state :runner laamb 2)
      (is (core/has-subtype? (refresh ichi) "Barrier") "Laamb should give Ichi the Barrier subtype")
      (card-ability state :runner engolo 2)
      (is (core/has-subtype? (refresh ichi) "Code Gate") "Engolo should give Ichi the Code Gate subtype"))))
```

**Pull request merged**